compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -150,14 +187,21 @@ impl<'context> Elaborator<'context> {
         }
 
         let span = path.span();
-        let mut args = vecmap(args, |arg| self.resolve_type_inner(arg));
 
         if let Some(type_alias) = self.lookup_type_alias(path.clone()) {
             let type_alias = type_alias.borrow();
             let expected_generic_count = type_alias.generics.len();
             let type_alias_string = type_alias.to_string();
             let id = type_alias.id;
 
+            let mut args = vecmap(type_alias.generics.iter().zip(args), |(generic, arg)| {
+                if generic.is_numeric_generic {
+                    self.resolve_numeric_type(arg)
+                } else {
+                    self.resolve_type(arg)
+                }
```

Comment:
```suggestion
                self.resolve_type_inner(arg, generic.kind())
```

Reply:
Applied this manually as `kind` was already public so I did not need a func